src/Symfony/Component/Validator/Constraints/DefaultPasswordStrengthEstimator.php

```diff
@@ -0,0 +1,52 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Validator\Constraints;
+
+final class DefaultPasswordStrengthEstimator
+{
+    /**
+     * @var non-empty-array{score: int, minEntropy: int}
+     */
+    private array $criteria = [
+        ['score' => 4, 'minEntropy' => 120],
+        ['score' => 3, 'minEntropy' => 100],
+        ['score' => 2, 'minEntropy' => 80],
+        ['score' => 1, 'minEntropy' => 60],
+        ['score' => 0, 'minEntropy' => 0],
+    ];
+
+    /**
+     * @param string[] $restrictedData
+     *
+     * @return non-empty-array{score: int, restrictedData: string[]}
+     */
+    public function __invoke(#[\SensitiveParameter] string $password, array $restrictedData): array
+    {
+        $wordsFound = [];
+        if (\count($restrictedData) > 0) {
+            preg_match_all('/'.implode('|', $restrictedData).'/i', $password, $wordsFound);
```

Comment:
this implementation will treat `$restrictedData` as being regex patterns, not static strings that are forbidden, as you don't escape them with `preg_quote`. This would be quite unexpected to me.

Reply:
Exact. I fixed it in the last commits